jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorUtil.kt

```diff
@@ -97,5 +97,11 @@ object CodeWhispererEditorUtil {
     /**
      * Checks if the [otherRange] overlaps this TextRange. Note that the comparison is `<` because the endOffset of TextRange is exclusive.
      */
-    fun TextRange.overlaps(otherRange: TextRange) = max(startOffset, otherRange.startOffset) < min(endOffset, otherRange.endOffset)
+    fun TextRange.overlaps(otherRange: TextRange): Boolean =
+        if (otherRange.isEmpty) {
+            // Handle case when otherRange is empty and within the range
+            otherRange.startOffset in startOffset until endOffset
+        } else {
+            max(startOffset, otherRange.startOffset) < min(endOffset, otherRange.endOffset)
+        }
```

Comment:
In what scenario do we have an empty range?

Reply:
When the new text is added, the `oldRange` is empty. `oldText` or `oldLength` refers to the text that was either removed or replaced. This will be empty range when the new text is added and nothing is removed/replaced.

Reply:
if users remove the whole recommendation, it will be empty